pkg/cleaner/aws/aws.go

```diff
@@ -60,78 +71,106 @@ func (a *Cleaner) Clean() error {
 		// a.cleanHostedZones,
 	}
 
+	errors := &errorcollection.ErrorCollection{}
+
 	for _, f := range cleaners {
+		a.logger.Log("level", "debug", "message", fmt.Sprintf("running cleaner %s", getFunctionName(f)))
 		err := f()
 		if err != nil {
-			return microerror.Mask(err)
+			a.logger.Log("level", "error", "message", fmt.Sprintf("running cleaner %s", getFunctionName(f)), "stack", fmt.Sprintf("%#v", err))
+			if val, ok := err.(*errorcollection.ErrorCollection); ok {
```

Comment:
Why do we need casting here ?
`Append` should be able to accept `err` since it satisfy `error`.

Reply:
True